cdap-app-fabric/src/test/java/co/cask/cdap/metadata/KafkaMetadataChangePublisherTest.java

```diff
@@ -0,0 +1,231 @@
+      new MetadataRecord(dataset, ImmutableMap.<String, String>of(), ImmutableList.<String>of()),
+      new MetadataRecord(dataset, ImmutableMap.of("key1", "v1"), ImmutableList.<String>of())
+    );
+    updateTime = currentTime - 600;
+    changesBuilder.add(new MetadataChangeRecord(previous, propRemoved, updateTime));
+    // Metadata after property deletion
+    previous = new MetadataRecord(dataset, ImmutableMap.of("key2", "value2"), ImmutableList.of("tag1"));
+    // Change 4: remove a tag - translates to 1 deletion
+    MetadataChangeRecord.MetadataDiffRecord tagRemoved = new MetadataChangeRecord.MetadataDiffRecord(
+      new MetadataRecord(dataset, ImmutableMap.<String, String>of(), ImmutableList.<String>of()),
+      new MetadataRecord(dataset, ImmutableMap.<String, String>of(), ImmutableList.of("tag1"))
+    );
+    updateTime = currentTime - 400;
+    changesBuilder.add(new MetadataChangeRecord(previous, tagRemoved, updateTime));
+    // Metadata after tag removal
+    previous = new MetadataRecord(dataset, ImmutableMap.of("key2", "value2"), ImmutableList.<String>of());
+    // Change 5: add a tag - translates to 1 addition
+    MetadataChangeRecord.MetadataDiffRecord tagAdded = new MetadataChangeRecord.MetadataDiffRecord(
+      new MetadataRecord(dataset, ImmutableMap.<String, String>of(), ImmutableList.of("tag1")),
+      new MetadataRecord(dataset, ImmutableMap.<String, String>of(), ImmutableList.<String>of())
+    );
+    updateTime = currentTime - 200;
+    changesBuilder.add(new MetadataChangeRecord(previous, tagAdded, updateTime));
+    return changesBuilder.build();
+  }
+
+  private void assertPublishedMetadata(List<MetadataChangeRecord> expected) throws InterruptedException {
+    String topic = cConf.get(Constants.Metadata.METADATA_UPDATES_KAFKA_TOPIC);
+    final CountDownLatch latch = new CountDownLatch(expected.size());
+    final CountDownLatch stopLatch = new CountDownLatch(1);
+    final List<MetadataChangeRecord> actual = new ArrayList<>(expected.size());
+    Cancellable cancellable = kafkaClient.getConsumer().prepare().addFromBeginning(topic, 0).consume(
+      new KafkaConsumer.MessageCallback() {
+        @Override
+        public void onReceived(Iterator<FetchedMessage> messages) {
+          while (messages.hasNext()) {
+            ByteBuffer payload = messages.next().getPayload();
+            MetadataChangeRecord record = GSON.fromJson(Bytes.toString(payload), MetadataChangeRecord.class);
+            actual.add(record);
+            latch.countDown();
+          }
+        }
+
+        @Override
+        public void finished() {
+          stopLatch.countDown();
+        }
+      }
+    );
+    Assert.assertTrue(latch.await(15, TimeUnit.SECONDS));
+    cancellable.cancel();
+    Assert.assertTrue(stopLatch.await(1, TimeUnit.SECONDS));
+    Assert.assertTrue(expected.containsAll(actual));
+    Assert.assertTrue(actual.containsAll(expected));
```

Comment:
The above two assertions can be replaced by `Assert.assertEquals(expected, actual)`, right?